### ISSUE TITLE
Deceased Known Relationships in Reports not working

### ISSUE THREAD

**Issue by karenn-ccbc** [SparkDevNetwork/Rock#5557]
### Please go through all the tasks below

- [X] Check this box only after you have successfully completed both the above tasks

### Please provide a brief description of the problem. Please do not forget to attach the relevant screenshots from your side.

In relation to item #4120 
The ability to "include deceased" in a dataview works and allows for known relationship of survivor to be used in reports. However, the reverse does not work. Pulling a report of Widows and wanting to show the inverse (deceased records) as a related person/known relationship in a report will not display that related person who is deceased.
image

image

### Expected Behavior

I would expect to be able to show the person name and relationship of the deceased person in the report.

### Actual Behavior

The related person is left blank.

### Steps to Reproduce

Create Known Relationships of Deceased and Survived By
Add a Deceased spouse as a known relationship to the surviving spouse
Run a Dataview of the Survivors, (ours is based on marital status of Widowed)
Add the Related People ( Deceased role) to the report
The report column is blank because the person record who is the relationship is deceased.
![image](https://github.com/SparkDevNetwork/Rock/assets/35667472/e2d01e1d-88b1-464c-b35c-dc2141b56b6d)
![image](https://github.com/SparkDevNetwork/Rock/assets/35667472/9144ae19-e6cd-40de-8361-cebb2f602e54)

### Rock Version

14.2

### Client Culture Setting

en-US

**Comment by karenn-ccbc** [comment]
Adding a comment from Rocketchat Jeff Richmond

It's possible that behavior is intentional, but at the very least it would be nice to have the option to include or exclude deceased people from the Related People field type.

**Comment by nairdo** [comment]
Hi @karenn-ccbc  That is the expected behavior, but your idea makes perfect sense.  

I would encourage you to submit it onto the http://community.rockrms.com/Ideas board.  It would probably end up as a new, optional checkbox on that ['Related People'](https://github.com/SparkDevNetwork/Rock/blob/82d738a0647c224002db9a352e24f0264012349a/Rock/Reporting/DataSelect/Person/RelatedPeopleSelect.cs#L429-L465) data-select-component:
![image](https://github.com/SparkDevNetwork/Rock/assets/383807/1949e90e-3a01-4ee3-8d04-f1823cad54a2)
